File: 8kyu/(8 kyu) Sort and Star/(8 kyu) Sort and Star.py
# 1 Plain solution
def two_sort(array):
    res = ""
    a = sorted(array)
    
    for c in a[0]:
        res += c + "***"
        
    return res[:-3]

# 2 Straightforward solution
def two_sort(array):
    x = [
        c for c in sorted(array)[0]
    ]
    
    return "***".join(x)

# 3 Optimized solution
def two_sort(array):
    return "***".join(
        list(sorted(array)[0])
    )

# 4 Clever solution
def two_sort(array):
    return "***".join(min(array))

# 5 Coding golf
two_sort = lambda a: "***".join(min(a))

File: 8kyu/(8 kyu) Sort and Star/(8 kyu) Sort and Star.dart
// 1 Simple solution
String twoSort(List<String> arr) {
  int size = arr.length;
  String result = "";
  for (int i = 0; i < size - 1; i++) {
    for (int j = i + 1; j < arr.length; j++) {
      if (arr[i].compareTo(arr[j]) > 0) {
        String temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
      }
    }
  }
  String word = arr[0];
  int length = word.length - 1;
  for (int i = 0; i < length; i++) {
    result += word[i] + "***";
  }
  return result + word[length];
}

// 2 Optimized solution
String twoSort(List<String> arr) {
  arr.sort();
  String word = arr[0];
  return word.split('').join('***');
}

// 3 Clever solution
String twoSort(List<String> arr) => (arr..sort()).first.split('').join('***');

// 4 Coding golf
String twoSort(List<String> a) => (a..sort())[0].split('').join('***');
